server/request.go

```diff
@@ -100,25 +100,83 @@ func (r *RpcRequest) process() {
 	r.body, err = ioutil.ReadAll(r.req.Body)
 	if err != nil {
 		r.logError("failed to read request body: %v", err)
-		r.writeHeaderStatus(http.StatusBadRequest)
+		r._writeHeaderStatus(http.StatusBadRequest)
 		return
 	}
 
 	if len(r.body) == 0 {
-		r.writeHeaderStatus(http.StatusBadRequest)
+		r._writeHeaderStatus(http.StatusBadRequest)
 		return
 	}
 
-	// Parse JSON RPC
-	if err = json.Unmarshal(r.body, &r.jsonReq); err != nil {
-		r.log("failed to parse JSON RPC request: %v - body: %s", err, r.body)
-		r.writeHeaderStatus(http.StatusBadRequest)
+	// Parse JSON RPC payload
+	if err = r.UnmarshalJSON(r.body); err != nil {
+		r.log("Parse payload %v", err)
```

Comment:
Perhaps send the error response here rather than inside `UnmarshalJSON`

Reply:
Agreed

Reply:
updated